Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesQuasi.py

```diff
@@ -407,7 +403,377 @@ def PyExec(self):
         if self._program == 'QL':
             self.setProperty('OutputWorkspaceProb', probWS)
 
+    def QLAddSampleLogs(self, workspace, fit_program, e_range, binning):
+
+        sample_binning, res_binning = binning
+        energy_min, energy_max = e_range
+
+        sample_logs = [('res_workspace', self._resWS),('fit_program', fit_program),('background', self._background),('elastic_peak', self._elastic),('energy_min', energy_min),('energy_max', energy_max),('sample_binning', sample_binning),('resolution_binning', res_binning)]
+
+        resnorm_used = (self._resnormWS != '')
+        sample_logs.append(('resnorm', str(resnorm_used)))
+        if resnorm_used:
+            sample_logs.append(('resnorm_file', str(self._resnormWS)))
+
+        width_file_used = (self._wfile != '')
+        sample_logs.append(('width', str(width_file_used)))
+        if width_file_used:
+            sample_logs.append(('width_file', str(self._wfile)))
+
+        log_alg = self.createChildAlgorithm('AddSampleLogMultiple', 0.9, 1.0, False)
+        log_alg.setProperty('Workspace', workspace)
+        log_alg.setProperty('LogNames', [log[0] for log in sample_logs])
+        log_alg.setProperty('LogValues', [log[1] for log in sample_logs])
+        log_alg.execute()
+
+
+    def C2Se(self, sname):
+        outWS = sname+'_Result'
+        asc = self.readASCIIFile(sname+'.qse')
+        var = asc[3].split()                            #split line on spaces
+        nspec = var[0]
+        var = ExtractInt(asc[6])
+        first = 7
+        Xout = []
+        Yf = []
+        Ef = []
+        Yi = []
+        Ei = []
+        Yb = []
+        Eb = []
+        ns = int(nspec)
+
+        dataX = np.array([])
+        dataY = np.array([])
+        dataE = np.array([])
+
+        for _ in range(0,ns):
+            first,Q,_,fw,it,be = self.SeBlock(asc,first)
+            Xout.append(Q)
+            Yf.append(fw[0])
+            Ef.append(fw[1])
+            Yi.append(it[0])
+            Ei.append(it[1])
+            Yb.append(be[0])
+            Eb.append(be[1])
+        Vaxis = []
+
+        dataX = np.append(dataX,np.array(Xout))
+        dataY = np.append(dataY,np.array(Yi))
+        dataE = np.append(dataE,np.array(Ei))
+        nhist = 1
+        Vaxis.append('f1.Amplitude')
+
+        dataX = np.append(dataX, np.array(Xout))
+        dataY = np.append(dataY, np.array(Yf))
+        dataE = np.append(dataE, np.array(Ef))
+        nhist += 1
+        Vaxis.append('f1.FWHM')
+
+        dataX = np.append(dataX,np.array(Xout))
+        dataY = np.append(dataY,np.array(Yb))
+        dataE = np.append(dataE,np.array(Eb))
+        nhist += 1
+        Vaxis.append('f1.Beta')
```

Comment:
appending in this way might look good as a refactored function. could also cut down on code duplication. function could be something like: `def _add_xye_data(self, data, axis_name, hist_number):` where data is a list `[x, y, e]`